test/pdevs_dynamic_atomic_test.cpp

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2013-2016, Laouen Mayal Louan Belloli
+ * Carleton University, Universidad de Buenos Aires
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#define BOOST_TEST_DYN_LINK
+#include <boost/test/unit_test.hpp>
+#include <cadmium/basic_model/accumulator.hpp>
+#include <cadmium/modeling/dynamic_atomic.hpp>
+
+/**
+  * This test is for the dynamic atomic class that wraps an atomic model to make it pointer friendly
+  */
+template<typename TIME>
+using int_accumulator=cadmium::basic_models::accumulator<int, TIME>;
+
+BOOST_AUTO_TEST_SUITE( pdevs_dynamic_atomic_test_suite )
+
+    BOOST_AUTO_TEST_CASE(create_dynamic_atomic_test) {
+        int_accumulator<float> model;
+        cadmium::modeling::dynamic_atomic<float, int_accumulator> wrapped_model;
+        BOOST_CHECK(typeid(model.state) == typeid(wrapped_model.state));
```

Comment:
It may be better to test with an static_assert and "is_same<>" here, it is only checking types.